modules/store/src/actions_subject.ts

```diff
@@ -13,12 +19,14 @@ export class ActionsSubject extends BehaviorSubject<Action>
   }
 
   next(action: Action): void {
-    if (typeof action === 'undefined') {
-      throw new TypeError(`Actions must be objects`);
+    if (typeOf(action) !== 'object') {
```

Comment:
would `typeof action !== 'object'` not work?

Reply:
There is a edge case when you pass accidently an array instead of object. 
```js
typeof [] // object
```
```js
({}).toString.call([]).slice(8, -1).toLowerCase() // array
```
I just wanted a strict treatment regardless of whether we typescript or not.

Reply:
It's true that array is type of `object` as well, however:
- we are trying to catch the case where `function` is passed instead of `object`, for example action creator is passed without calling it
- even if array doesn't run into this check, it will be caught with the next if-statement `typeof action.type === 'undefined'`

Also, I find the code snippet `({}).toString.call([]).slice(8, -1).toLowerCase()` to be somewhat "crafty" 😁and not very straightforward/readably. I'd rather avoid it.

Reply:
I agree with @alex-okrushko.
To keep it readable I'm in favor of using `typeof`.

Plus we already have a compile error if we try to dispatch an array. The `createAction` is only the problem here.

![image](https://user-images.githubusercontent.com/28659384/59051220-9bab3b00-888c-11e9-9f44-03e84702a38b.png)

Reply:
I had the same thoughts (compile error) after reading what @alex-okrushko wrote. I was to strict! I have to find a balance between typescript and javascript :)

Reply:
`({}).toString.call([]).slice(8, -1).toLowerCase()` its removed. There is now a simple the for function.